extensions/flags/js/src/forum/extend.ts

```diff
@@ -1,4 +1,12 @@
 import Extend from 'flarum/common/extenders';
+import Post from 'flarum/common/models/Post';
 import FlagsPage from './components/FlagsPage';
+import Flag from './models/Flag';
 
-export default [new Extend.Routes().add('flags', '/flags', FlagsPage)];
+export default [
+  new Extend.Routes().add('flags', '/flags', FlagsPage),
+
+  new Extend.Model(Flag).register('flags'),
```

Comment:
Does this mean that one class could be registered for multiple models?

Would attributes / relationships registered on a model be set for all the name bindings it's associated with?

Reply:
It's a translation of `app.store.models[type] = class;`, and since registering attributes and relationships adds to the class prototype then it would be the case yeah, one could use a model for multiple resource types, although I cat think of a valid use case for that.

Reply:
I see where you're coming from. Since the goal here is to add to the store the model class that maps from the model type string `'flags' => Flag` we could either have it as an optional constructor argument, or maybe even better, within a `Store` extender.

```ts
new Store().add('flags', Flag),
```